src/cl.jl

```diff
@@ -112,3 +108,183 @@ function cl2dl(cl::AbstractVector{T}, lmin::Integer) where {T <: Real}
+
+# ARGUMENTS
+- `cl::AbstractVector{T}`: The array representing the power spectrum components ``C_{\\ell}``,
+starting from `` \\ell = 0 ``.
+- `lmax::Integer`: the maximum ``ℓ`` coefficient, will default to `length(cl)-1` if not specified.
+- `mmax::Integer`: the maximum ``m`` coefficient, will default to `lmax` if not specified.
+- `rng::AbstractRNG` : (optional) the RNG to be used for generating the ``a_{\\ell m}``. It allows
+to set the seed beforehand guaranteeing the reproducibility of the process.
+"""
+function synalm(cl::Vector{T}, lmax::Integer, mmax::Integer, rng::AbstractRNG) where {T <: Real}
+    cl_size = length(cl)
+    (cl_size - 1 >= lmax) || throw(DomainError(cl_size, "not enough C_l's to generate Alm"))
+    alm = Alm{ComplexF64, Vector{ComplexF64}}(lmax, mmax)
+    synalm!(cl, alm, rng)
+    alm
+end
+
+synalm(cl::Vector{T}, lmax::Integer, mmax::Integer) where {T <: Real} =
+    synalm(cl, lmax, mmax, Random.seed!(1234))
+
+synalm(cl::Vector{T}, lmax::Integer, rng::AbstractRNG) where {T <: Real} =
+    synalm(cl, lmax, lmax, rng)
+
+synalm(cl::Vector{T}, lmax::Integer) where {T <: Real} =
+    synalm(cl, lmax, lmax, Random.seed!(1234))
+
+synalm(cl::Vector{T}, rng::AbstractRNG) where {T <: Real} =
+    synalm(cl, length(cl) - 1, length(cl) - 1, rng)
+
+synalm(cl::Vector{T}) where {T <: Real} =
+    synalm(cl, length(cl) - 1, length(cl) - 1, Random.seed!(1234))
+
+#########################################################################
+
+"""
+    synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}, lmax::Integer, rng::AbstractRNG) where {T <: Real}
+    synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}, lmax::Integer) where {T <: Real}
+    synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}, rng::AbstractRNG) where {T <: Real}
+    synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}) where {T <: Real}
+
+Generate a map from a given power spectra ``C_{\\ell}``. The result is saved into
+the `HealpixMap` passed in input.
+
+# ARGUMENTS
+- `cl::AbstractVector{T}`: The array representing the power spectrum components ``C_{\\ell}``.
+- `map::HealpixMap{T, RingOrder}`: the map that will contain the result.
+- `lmax::Integer`: the maximum ``ℓ`` coefficient, will default to `length(cl)-1` if not specified.
+- `rng::AbstractRNG` : (optional) the RNG to be used for generating the ``a_{\\ell m}``. It allows
+to set the seed beforehand guaranteeing the reproducibility of the process.
+"""
+function synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}, lmax::Integer, rng::AbstractRNG) where {T <: Real}
+    cl_size = length(cl)
+    (cl_size - 1 >= lmax) || throw(DomainError(cl_size, "not enough C_l's to generate Alm"))
+    alm = Alm{ComplexF64, Vector{ComplexF64}}(lmax, lmax)
+    synalm!(cl, alm)
+    alm2map!(alm, map)
+end
+
+synfast!(cl::Vector{T}, map::HealpixMap{T, RingOrder}, lmax::Integer) where {T <: Real} =
+    synfast!(cl, map, lmax, Random.seed!(1234))
```

Comment:
This implementation always uses the same RNG seed, which means that two consecutive calls will produce exactly the same a_ℓm values. Is this intended?

I believe a better approach would be to use the global RNG `Random.GLOBAL_RNG`. In this way, calling `synfast!` twice will just reuse the generator and produce two different realizations, which is probably what 99% of the users want.

(The same probably applies to other instances below).

Reply:
Right!! fixed now.